h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -831,17 +831,25 @@ private FVecParseWriter makeDout(ParseSetup localSetup, int chunkOff, int nchunk
           // Zipped file; no parallel decompression;
           InputStream bvs = vec.openStream(_jobKey);
           ZipInputStream zis = new ZipInputStream(bvs);
-          ZipEntry ze = zis.getNextEntry(); // Get the *FIRST* entry
+          ZipEntry ze = null;
+
+          if (ZipUtil.isZipDirectory(key)) {  // file is a zip if multiple files
+            ze = zis.getNextEntry();          // first ZipEntry describes the directory
+          }
+
+          ze = zis.getNextEntry(); // Get the *FIRST* entry
           // There is at least one entry in zip file and it is not a directory.
           if( ze != null && !ze.isDirectory() )
             _dout[_lo] = streamParse(zis,localSetup, makeDout(localSetup,chunkStartIdx,vec.nChunks()), bvs);
             _errors = _dout[_lo].removeErrors();
+
```

Comment:
Is this dead code? I don't see where ze2 is used. Also it looks like the handling of multiple files was moved inside streamParse.